pkg/storage/unified/resource/server.go

```diff
@@ -29,6 +29,9 @@ type ResourceServer interface {
 	ResourceIndexServer
 	BlobStoreServer
 	DiagnosticsServer
+
+	// Initialize the server before any traffic is served
+	Init(ctx context.Context) error
```

Comment:
how do we want to expose this?

Reply:
what do you mean?

Reply:
Do we want to make this part of this interface? Do we not want to expose it and init it within `NewResourceServer()`? Still unsure. Right now just want to see what this might break.